packages/ssx-sdk/src/modules/Storage.ts

```diff
@@ -1,42 +1,278 @@
+      upgrade(db) {
+        if (!db.objectStoreNames.contains(this.storeName)) {
+          db.createObjectStore(this.storeName);
+        }
+      },
+    });
+  }
+
+  public async get(key: string): Promise<any> {
+    const db = await this.getDB();
+    const tx = db.transaction(this.storeName, 'readonly');
+    const store = tx.objectStore(this.storeName);
+    const value = await store.get(this.prefixKey(key));
+    await tx.done;
+    return value;
+  }
+
+  public async put(key: string, value: any): Promise<void> {
+    const db = await this.getDB();
+    const tx = db.transaction(this.storeName, 'readwrite');
+    const store = tx.objectStore(this.storeName);
+    await store.put(value, this.prefixKey(key));
+    await tx.done;
+  }
+
+  public async list(): Promise<string[]> {
+    const db = await this.getDB();
+    const tx = db.transaction(this.storeName, 'readonly');
+    const store = tx.objectStore(this.storeName);
+    const keys = await store.getAllKeys();
+    await tx.done;
+    return keys as unknown as string[];
+  }
+
+  public async delete(key: string): Promise<void> {
+    const db = await this.getDB();
+    const tx = db.transaction(this.storeName, 'readwrite');
+    const store = tx.objectStore(this.storeName);
+    await store.delete(this.prefixKey(key));
+    await tx.done;
+  }
+
+  public async deleteAll(): Promise<void> {
+    const db = await this.getDB();
+    const tx = db.transaction(this.storeName, 'readwrite');
+    const store = tx.objectStore(this.storeName);
+    await store.clear();
+    await tx.done;
   }
 }
-class KeplerDataVault implements IDataVault {
-  private userAuth: IUserAuthorization;
-  private encryption: IEncryption;
 
-  constructor(
-    config: any,
-    userAuth: IUserAuthorization,
-    encryption: IEncryption
-  ) {
-    this.userAuth = userAuth;
+/**
+ * Represents a class for managing browser storage operations with encryption.
+ */
+class BrowserDataVault extends BrowserStorage implements IDataVault {
+  private encryption: IEncryption;
+  /**
+   * Constructs a new instance of BrowserDataVault.
+   * @param config - An object containing optional configuration properties for browser storage.
+   * @param encryption - An instance of an object that implements IEncryption for encryption and decryption operations.
+   */
+  constructor(config: any, encryption: IEncryption) {
+    super(config);
     this.encryption = encryption;
   }
+
+  /**
+   * Retrieves the encrypted stored value for the specified key,
+   * decrypts it, and returns the decrypted value.
+   * @param key - The unique identifier for the stored value.
+   * @returns A Promise that resolves to the decrypted value associated
+   * with the given key or null if the key does not exist.
+   */
+  public async get(key: string): Promise<any> {
+    const encryptedData = await super.get(key);
+    if (encryptedData) {
+      return this.encryption.decrypt(encryptedData);
+    }
+    return null;
```

Comment:
maybe a better approach would be throwing an error instead of returning `null`

Reply:
Leaving as null because a few key-storage solution (like localstorage and AWS's S3 API) do this